controllers/pagerdutyintegration/handleupdate.go

```diff
@@ -28,20 +33,33 @@ func (r *PagerDutyIntegrationReconciler) handleUpdate(pdclient pd.Client, pdi *p
 		// SERVICE_ID and INTEGRATION_ID
 		configMapName string = config.Name(pdi.Spec.ServicePrefix, cd.Name, config.ConfigMapSuffix)
 	)
-
-	pdData, err := pd.NewData(pdi, cd.Spec.ClusterMetadata.ClusterID, cd.Spec.BaseDomain)
-	if err != nil {
-		return err
-	}
-	err = pdData.ParseClusterConfig(r.Client, cd.ObjectMeta.Namespace, configMapName)
+	cm := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: cd.ObjectMeta.Namespace, Name: configMapName}, cm)
 	if err != nil {
-		return err
+		return nil // requeue and wait for the configmap to be created
 	}
-	if pdData.AlertGroupingType != pdi.Spec.AlertGroupingParameters.Type || pdData.AlertGroupingTimeout != pdi.Spec.AlertGroupingParameters.Config.Timeout {
+
+	alertGroupingType := cm.Data["ALERT_GROUPING_TYPE"]
+	alertGroupingTimeout := cm.Data["ALERT_GROUPING_TIMEOUT"]
+
+	if alertGroupingType != pdi.Spec.AlertGroupingParameters.Type || alertGroupingTimeout != fmt.Sprintf("%d", pdi.Spec.AlertGroupingParameters.Config.Timeout) {
+		pdData, err := pd.NewData(pdi, cd.Spec.ClusterMetadata.ClusterID, cd.Spec.BaseDomain)
+		if err != nil {
+			return err
+		}
+		err = pdData.ParseClusterConfig(r.Client, cd.ObjectMeta.Namespace, configMapName)
+		if err != nil {
+			return err
+		}
+
 		err = pdclient.UpdateAlertGrouping(pdData)
 		if err != nil {
 			return err
 		}
+
+		cm.Data["ALERT_GROUPING_TYPE"] = pdi.Spec.AlertGroupingParameters.Type
+		cm.Data["ALERT_GROUPING_TIMEOUT"] = fmt.Sprintf("%d", pdi.Spec.AlertGroupingParameters.Config.Timeout)
+		r.Client.Update(context.TODO(), cm)
```

Comment:
We should handle the error for this one too to make sure it gets requeued if this fails